userspace/libsinsp/user_event.cpp

```diff
@@ -266,18 +371,19 @@ sinsp_user_event& sinsp_user_event::operator=(sinsp_user_event&& other)
 std::string sinsp_user_event::to_string(uint64_t timestamp,
 										std::string&& name,
 										std::string&& description,
-										std::string&& scope,
+										event_scope&& scope,
 										tag_map_t&& tags,
 										uint32_t sev)
 {
 	const std::string from("\"");
 	const std::string to("\\\"");
 
+	std::string s(scope.get());
```

Comment:
This breaks the move optimization by requiring a copy from scope to the local string, but you can avoid the copy by changing get() to return an rvalue reference of event_scope::m_scope.

The rvalue reference version could also use a scarier name than get() so it's clear it's moving the string.